src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -245,16 +237,20 @@ private static boolean shouldBeOnSameLine(RightCurlyOption bracePolicy, Details
      * @param bracePolicy option for placing the right curly brace
      * @param details Details for validation
      * @return true if a right curly should be alone on a line.
+     * @noinspection OverlyComplexBooleanExpression
      */
+    // -@cs[BooleanExpressionComplexity] keeping logic in single method makes it clearer
```

Comment:
I did think about it, but
1) it is actually better to see all the logic in one place - it becomes clearer
2) the best option of splitting I could imagine was smth like:
```java
return shouldBeAloneOnLineWithAloneOption(details)
               && bracePolicy == RightCurlyOption.ALONE
           || shouldBeAloneOnLineWithAloneOrSingleLineOption(details)
               && bracePolicy == RightCurlyOption.ALONE_OR_SINGLELINE
           || details.shouldCheckLastRcurly
               && details.rcurly.getLineNo() == details.nextToken.getLineNo();
```